journal.py

```diff
@@ -0,0 +1,214 @@
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
+    )
+
+    def __repr__(self):
+        return "Entry: {title} created at {date}".format(
+            title=self.title, date=self.created
+        )
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.created.desc()).all()
+
+    @classmethod
+    def by_id(cls, id_, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).filter(cls.id == id_).first()
+
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, text=text)
+        session.add(instance)
+        return instance
+
+
+def init_db():
+    engine = sa.create_engine(DATABASE_URL, echo=False)
+    Base.metadata.create_all(engine)
+
+
+def do_login(request):
+    username = request.params.get('username', None)
+    password = request.params.get('password', None)
+    if not (username and password):
+        raise ValueError('both username and password are required')
+
+    settings = request.registry.settings
+    manager = BCRYPTPasswordManager()
+    if username == settings.get('auth.username', ''):
+        hashed = settings.get('auth.password', '')
+        return manager.check(hashed, password)
+    return False
+
+
+@view_config(route_name='home', renderer='templates/list.jinja2')
+def list_view(request):
+    entries = Entry.all()
+    return {'entries': entries}
+
+
+@view_config(route_name='about', renderer='templates/about.jinja2')
+def about_view(request):
+    return {}
+
+
+@view_config(route_name='detail', renderer='templates/detail.jinja2')
+def detail_view(request):
+    entry_id = int(request.matchdict.get('id', -1))
+    entry = Entry.by_id(entry_id)
+    if entry is None:
+        return HTTPNotFound()
+    return {'entry': entry}
+
+
+@view_config(route_name='edit', renderer='templates/edit.jinja2')
+def edit_view(request):
+    if not request.authenticated_userid:
+        return HTTPFound(request.route_url('login'))
+    entry_id = int(request.matchdict.get('id', -1))
+    if request.method == 'POST':
+        return HTTPFound(request.route_url('detail', id=entry_id))  # Add edit
+    entry = Entry.by_id(entry_id)
+    if entry is None:
+        return HTTPNotFound()
+    return {'entry': entry}
+
+
+@view_config(route_name='create', renderer='templates/create.jinja2')
+def create_view(request):
+    if not request.authenticated_userid:
+        return HTTPFound(request.route_url('login'))
+    return {}
+
+
+@view_config(route_name='add', request_method='POST')
+def add_entry(request):
+    title = request.params.get('title')
+    text = request.params.get('text')
+    Entry.write(title=title, text=text)
+    return HTTPFound(request.route_url('home'))
```

Comment:
Can you combine the `create_view` and `add_entry` views into one view function using the `if request.method == 'POST'` approach from the `edit_view` above?